src/pages/ProductList/index.jsx

```diff
@@ -1,122 +1,128 @@
 import ProductSection from '../../components/ProductSection';
-import Colors from '../../utils/colors';
-import Spiner from './components/Loader';
-
-
-const Container = styled.div`
-  display: flex;
-  margin-right: 8%;
-  .container__sidebar {
-    width: ${(props) => props.collapsed ? '4.25rem' : '36%'};
-    background-color: white;
-    overflow-y: hidden;
-  }
-  .container__content {
-    background-color: white;
-    padding: 1rem;
-    margin-left: 1rem;
-    width: -webkit-fill-available;
-  }
-`;
-
-const Sidebar = styled.section`
-  .sidebar__section {
-    display: flex;
-    align-items: center;
-  }
-  .sidebar__icon {
-    min-width: 4rem;
-    width: 4rem;
-    height: 5rem;
-    display: flex;
-    justify-content: center;
-    align-items: center;
-    img {
-      width: 2.5rem;
-      height: 2.5rem;
-    }
-  }
-  .check {
-    height: 3rem;
-  }
-  .sidebar__bullet {
-    width: 0.6rem;
-    height: 0.6rem;
-    background-color: black;
-    border-radius: 50%;
-  }
-`;
-
-const Pagination = styled.section`
-  nav {
-    margin: 3rem 0 2rem 0;
-    display: flex;
-    justify-content: right;
-  }
-  a {
-    min-width: 2rem;
-    border: 0.2rem solid ${Colors['BG-C']};
-    cursor: pointer;
-    font-size: 1.8rem;
-    padding: 0.5rem 1.5rem;
-    text-decoration: none;
-    color: black;
-    border-radius: 0.5rem;
-    &:hover {
-      background-color: brown;
-      color: white;
-    }
-  }
-`;
-
-const NoProducts = styled.div`
-  height: 8rem;
-  display: flex;
-  justify-content: center;
-  align-items: center;
-`;
+import { useFeaturedCategories } from '../../utils/hooks/useFeaturedCategories';
+import Loader from '../../components/Loader';
+import { Container, Sidebar, Pagination, NoProducts, ClearButton } from './styles';
+import { useFetching } from '../../utils/hooks/useFetch';
+const URL = 'https://wizeline-academy.cdn.prismic.io/api/v2/documents/search?ref=YZaBvBIAACgAvnOP&q=%5B%5Bat%28document.type%2C+%22product%22%29%5D%5D&pageSize=12&languageCode=en-us&page=';
 
 
 const ProductList = () => {
   const [isMenuCollapse,setIsMenuCollapse] = useState(false);
   const [activeFilters, setActiveFilters] = useState([]);
-  const [categoryList, setCategoryList] = useState([]);
   const [products, setProducts] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
+  const [pages, setPages] = useState([]);
+  const categoryList = useFeaturedCategories().data.results;
+  const [fetchProducts, setFetchProducts] = useState([]);
+  const [nextPage, setNextPage] = useState('');  
+  const [previousPage, setPreviousPage] = useState(''); 
+  // const [totalPages, setTotalPages] = useState(1);
+  let firstCall = useFetching(URL+1);  
 
   useEffect(() => {
-    setCategoryList([...CategoryList['results']]);
-    setProducts([...productList['results']]);
-    setTimeout(() => {
-      setIsLoading(false);
-    }, 2000);
-  }, []);
+    const data = firstCall?.data;
```

Comment:
<img width="1607" alt="image" src="https://user-images.githubusercontent.com/26983421/161099704-61c0de24-975a-4280-84d8-c2724180b4ec.png">
When I click view all products, seems the same endpoint API is being called twice, this could be a performance leek